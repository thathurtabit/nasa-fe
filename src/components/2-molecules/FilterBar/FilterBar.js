import React from 'react';
import PropTypes from 'prop-types';
import uuidv1 from 'uuid/v1';
import { connect } from 'react-redux';
import { mediaType } from '../../../utils/constants/constants';
import FilterBarStyled, { Filter } from './FilterBar.styled';
import { setFilter } from '../../../state/actions/setFilter';
import { setSearch } from '../../../state/actions/setSearch';

const mapStateToProps = state => ({
  filteredMediaType: state.filteredMediaType,
  searchValue: state.searchValue,
});

const mapDispatchToProps = dispatch => ({
  setFilter: type => dispatch(setFilter(type)),
  setSearch: searchObj => dispatch(setSearch(searchObj)),
});

const FilterBar = ({
  filteredMediaType,
  setFilter,
  setSearch,
  searchValue,
}) => {
  const handleFilterClick = e => {
    const clickValue = e.target.textContent;

    setFilter(filteredMediaType === clickValue ? '' : clickValue);
    setSearch({
      search: searchValue.search,
      type: filteredMediaType === clickValue ? '' : clickValue,
    });
  };

  return (
    <FilterBarStyled onClick={e => handleFilterClick(e)}>
      {Object.keys(mediaType).map(type => (
        <Filter
          key={uuidv1()}
          title={`Filter results by ${type}`}
          active={filteredMediaType === type}
        >
          {type}
        </Filter>
      ))}
    </FilterBarStyled>
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilterBar);

FilterBar.propTypes = {
  setSearch: PropTypes.func.isRequired,
  setFilter: PropTypes.func.isRequired,
  searchValue: PropTypes.objectOf(PropTypes.string).isRequired,
  filteredMediaType: PropTypes.string.isRequired,
};
