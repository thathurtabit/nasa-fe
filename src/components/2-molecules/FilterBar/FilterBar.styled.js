import styled from 'styled-components';

export const Filter = styled.button`
  position: relative;
  background: ${props => (props.active ? 'red' : 'transparent')};
`;

const FilterBarStyled = styled.section`
  position: relative;
`;

export default FilterBarStyled;
