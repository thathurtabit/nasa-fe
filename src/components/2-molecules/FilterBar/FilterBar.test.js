import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import FilterBar from './FilterBar';

describe('FilterBar', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<FilterBar />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
