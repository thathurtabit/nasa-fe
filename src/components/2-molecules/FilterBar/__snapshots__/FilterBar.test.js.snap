// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
